### ISSUE TITLE
Specifying spec file fails to over-ride auto-generated legend

### ISSUE THREAD

**Issue by ibrewster** [GenericMappingTools/gmt#6641]
**Description of the problem**
According to the documentation for Figure.Legend:

> spec (None or [str](https://docs.python.org/3/library/stdtypes.html#str)) – Either None [default] for using the automatically generated legend specification file, or a filename pointing to the legend specification file.

Which would imply to me that if I give a filename, it should use that file _instead of_ the auto-generated legend. However, this does not appear to happen - the auto-generated legend is used instead.

**Full code that generated the error**

```python
import pygmt

bounds = [-180.7568317475191,
          -178.59975221352386,
          51.37616035103776,
          52.375229777638786]

fig = pygmt.Figure()
fig.basemap(projection="M10i", region=bounds, frame=('WeSn', 'afg'))

sta_x = -179
fig.plot(x = [sta_x, ], y = [52, ], style = 'a16p', color = 'red', label = "Wrong_1")
sta_x = 179.75
fig.plot(x = [sta_x, ], y = [52, ], style = 't16p', color = 'green', label = "Wrong_2")

pos = 'JTR+jTR+o0.2c+l1.5'

test_file = "/Users/israel/Desktop/legend.txt"

# Specifying a file as the first argument should over-ride the auto legend.
fig.legend(test_file, position = pos, box = "+gwhite+p1p", verbose = True)

fig.show(method = "external")
```
Where the contents of `/Users/israel/Desktop/legend.txt` are:
```
S - kvolcano 0.222222i red - - Plot1
S - kvolcano 0.222222i green - - Plot2
```
This produces the following image:
[7c3501330b2748ab827da002c428cde0.pdf](https://github.com/GenericMappingTools/pygmt/files/8393254/7c3501330b2748ab827da002c428cde0.pdf)

Removing the `label` argument from both plot commands allows the legend to properly use the file as expected.

**Full error message**

No error message is generated, however the verbose output of `legend` indicates it is using the auto-generated file rather than the user-specified one:

```
legend [INFORMATION]: Processing input text table data
legend [INFORMATION]: Processing hidden legend specification file /Users/israel/.gmt/sessions/gmt_session.22260/gmt.1.legend
legend [INFORMATION]: Reading Data Table from File /Users/israel/.gmt/sessions/gmt_session.22260/gmt.1.legend
legend [INFORMATION]: Central meridian not given, default to -179.678
legend [INFORMATION]: Map scale is 9.45374 km per cm or 1:945374.
legend [INFORMATION]: Legend height not given, using estimated height of 1.40367 cm.
plot [INFORMATION]: Processing input table data
plot [INFORMATION]: Map scale is 0.001 km per inch or 1:39.3701.
plot [INFORMATION]: Reading Data Table from memory reference supplied by pointer
text [INFORMATION]: Processing input text table data
text [INFORMATION]: Map scale is 0.000393701 km per cm or 1:39.3701.
text [INFORMATION]: Reading Data Table from memory reference supplied by pointer
text [INFORMATION]: pstext: Plotted 2 text strings
```

**System information**

Please paste the output of `python -c "import pygmt; pygmt.show_versions()"`:

```
import pygmt

bounds = [-180.7568317475191,
          -178.59975221352386,
          51.37616035103776,
          52.375229777638786]

fig = pygmt.Figure()
fig.basemap(projection="M10i", region=bounds, frame=('WeSn', 'afg'))

sta_x = -179
fig.plot(x = [sta_x, ], y = [52, ], style = 'a16p', color = 'red', label = "Wrong_1")
sta_x = 179.75
fig.plot(x = [sta_x, ], y = [52, ], style = 't16p', color = 'green', label = "Wrong_2")

pos = 'JTR+jTR+o0.2c+l1.5'

test_file = "/Users/israel/Desktop/legend.txt"

# Specifying a file as the first argument should over-ride the auto legend.
fig.legend(test_file, position = pos, box = "+gwhite+p1p", verbose = True)

fig.show(method = "external")
```

**Comment by PaulWessel** [comment]
Sorry, took me a while to look at this.  I looked at the legend documentation and I cannot find anywhere that says the spec file replaces entries created via -l.  The documentation says -M controls if both file and hidden info is used or just hidden [default].  Nothing about ignoring hidden info.  Why use -l if you are not going to use it?  We could certainly add a flag to -M to mean that, but is that really a good system?

**Comment by ibrewster** [comment]
My apologies then. I filed this because I found the behavior unexpected and unintuitive, but you are correct about the documentation not indicating that it should behave otherwise from how it does.

Apparently, however, it's just my expectations that are wrong. Lesson learned. I'll avoid filing reports in the future. Sorry about the noise!

**Comment by PaulWessel** [comment]
Well, the guiding principle of GMT has always be "it ought to be possible to do X".  It is easy for me to enhance **-M** so that we can handle all permutations:

1. Only place legend entries set with **-l**
2. Only place legend entries given by _specfile_
3. Append legend entries from _specfile_ to those set by **-l**
4. Append legend entries set by **-l** to those given by _specfile_

I think those are the possibilities, no?

**Comment by PaulWessel** [comment]
I am OK with that.

**Comment by PaulWessel** [comment]
We should check if changing this default has backwares compatibility consequences.

**Comment by PaulWessel** [comment]
OK, I got this now in a to-be-submitted-soon PR. Does it meet your expectations?  All subplots do
```
echo 2 2 | gmt plot -Sc0.2c -Gblack -lLABEL1
echo 5 5 | gmt plot -St0.2c -Gred -lLABEL2
gmt legend legend.txt -DjTL -F
```

and 5 of them have a -M option added as indicated in the tag

![mixlegend](https://user-images.githubusercontent.com/26473567/165979228-1ab93ee7-18f7-4fb1-b34c-60ecbe8385e0.png)

**Comment by PaulWessel** [comment]
OK, merged in #6644 so closing this issue.

**Comment by ibrewster** [comment]
Looks awesome, thanks!
